src/containers/setContacts/index.tsx

```diff
@@ -0,0 +1,202 @@
+import { Alert, Spin } from 'antd';
+import moment from 'moment';
+import React, { useEffect } from 'react';
+import { connect, useDispatch } from 'react-redux';
+import { useHistory, useLocation } from 'react-router';
+import protectedApiClient from '../../api/protectedApiClient';
+import { Routes } from '../../App';
+import ContactsForm from '../../components/ContactsForm';
+import { C4CState } from '../../store';
+import {
+  asyncRequestIsComplete,
+  asyncRequestIsFailed,
+} from '../../utils/asyncRequest';
+import { encodeProfileFieldFile } from '../../utils/fileEncoding';
+import {
+  participatingFamilySearchQuery,
+  participatingFamilySearchQueryFlag,
+} from '../../utils/signupFlow';
+import {
+  getContactInfo,
+  resetSetContactState,
+  setContactInfo,
+} from './ducks/thunks';
+import {
+  AdditionalContact,
+  Child,
+  ContactFormFields,
+  ContactInfo,
+  ContactsReducerState,
+} from './ducks/types';
+
+interface SetContactsProps {
+  readonly contacts: ContactsReducerState['contacts'];
+  readonly setContacts: ContactsReducerState['setContacts'];
+}
+
+const SetContacts: React.FC<SetContactsProps> = ({ contacts, setContacts }) => {
+  const dispatch = useDispatch();
+  const history = useHistory();
+  const location = useLocation();
+  useEffect(() => {
+    dispatch(getContactInfo());
+  }, [dispatch]);
+  const registeringAsParticipatingFamily =
+    new URLSearchParams(location.search).get(
+      participatingFamilySearchQueryFlag,
+    ) != null;
+
+  if (asyncRequestIsComplete(setContacts) && registeringAsParticipatingFamily) {
+    protectedApiClient.makePFRequest();
+    history.push({
+      pathname: Routes.SIGNUP_CONFIRMATION,
+      search: participatingFamilySearchQuery,
+    });
+  }
+
+  function selectProfilePictureOrDefaultToUndefined(
+    i: number,
+    arr: Child[] | AdditionalContact[],
+  ): string | undefined {
+    if (asyncRequestIsComplete(contacts) && i < arr.length) {
+      return arr[i].profilePicture;
+    } else {
+      return undefined;
+    }
+  }
+
+  const handleCloseSuccessAlert = () => {
+    dispatch(resetSetContactState());
+  };
+
+  const onFinish = async (values: ContactFormFields) => {
+    if (!asyncRequestIsComplete(contacts)) {
```

Comment:
What should be done here? Throwing an error will do nothing I think. Currently error is linked to the contacts redux, maybe we can use useState and have the error display if a custom error or the redux error appears?

Reply:
Wait, won't this always throw an error? the AsyncRequest `contacts` will not be complete when onFinish is called?

Reply:
contacts should be finished but setContacts will be pending or resolved